### PR TITLE
MATL: sync up funcs for both games

### PR DESCRIPTION
**Pull request by IJARika**
json parsing is now packed into mostly one spot (good for replacing stuff later), and funcs are mostly identical in code and features.

### CODE REVIEW

File: src/assets/material.cpp [hunk 1]
```diff
@@ -2,15 +2,11 @@
...
+void AutoAddMaterialAssetTextures(CPakFile* pak, std::vector<PakAsset_t>* assetEntries, rapidjson::Value& mapEntry)
 {
-    Log("Adding matl asset '%s'\n", assetPath);
-
-    // we need to take better account of textures once asset caching becomes a thing
-    short externalDependencyCount = 0;
-
-    if (mapEntry.HasMember("textures") && mapEntry["textures"].IsArray())
+    // cut if statement?
+    if (JSON_IS_STR(mapEntry, "textures"))
```
Comment (r-ex) [inline src/assets/material.cpp:9]: i think this should be JSON_IS_ARRAY

File: src/assets/material.cpp [hunk 2]
```diff
@@ -22,116 +18,344 @@ void Assets::AddMaterialAsset_v12(CPakFile* pak, std::vector<PakAsset_t>* assetE
...
+
+    // optional shaderset override
+    if (JSON_IS_STR(mapEntry, "shaderset"))
+    {
+        std::string shadersetPath = "shaderset/" + mapEntry["shaderset"].GetStdString() + ".rpak";
+        matl->shaderSet = RTech::GetAssetGUIDFromString(shadersetPath.c_str());
+    }
+
+    
+    // texan
```
Comment (r-ex) [inline src/assets/material.cpp:192]: texan

**Review by r-ex — COMMENTED** [review]
might have missed something else but otherwise looks good

File: src/assets/material.cpp (hunk 2 above)
Comment (IJARika) [reply src/assets/material.cpp:196]: texan

**Comment by r-ex** [comment]
thank

**Pull request merged**